src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -1983,10 +2083,30 @@ instance Semigroup LimitClause where
 
 instance Monoid LimitClause where
   mempty = Limit mzero mzero
-  mappend = (<>)
 
 -- | A locking clause.
-type LockingClause = Monoid.Last LockingKind
+data LockingClause =
+  GeneralLockingClause LockingKind
+  -- ^ Locking clause not specific to any database implementation
+  | PostgresLockingClause PostgresLockingKind
+  -- ^ Locking clause specific to postgres
```

Comment:
Ahh, so that's why we can't hoist the `PostgresLockingKind` to the PG module 🤔 

I think that's probably fine - `MySQL`'s locking clause can be different and in here, too, if we need to create it.